HPCC-16262 DOCS:Session Management

g-pan opened the pull request:
Fix HPCC-16262 DOCS:Session Management
Document new HPCC Session Management, also address HPCC-18904


@JamesDeFabia please review
@RussWhitehead please review

 
NOTE: For the file TheECLWatchMan.xml only material relevant to this issue is lines 124-153
(rest was inadvertently marked as changed after insertion of DNT tags for internationalization)   
 

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

@JamesDeFabia please review
@RussWhitehead please review

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

Please see successful Doc test build: http://10.240.32.243/view/Docs%207.0/job/DocDev70/

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

g-pan committed:
HPCC-16262 DOCS:Session Management

Fix HPCC-16262 DOCS:Session Management
Document new HPCC Session Management...
also address HPCC-18904

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: d3d2e5800f4e08b88a212707adfb597d969450b1
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   11 sec (00:00:11) |  533 sec (00:08:53) |  163 sec (00:02:43) |    6 sec (00:00:06) |   21 sec (00:00:21) | 1132 sec (00:18:52) |   16 sec (00:00:16) | 1882 sec (00:31:22) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 07b510781fc44a64a90d5434d8433ec0a3e0a1cf
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   14 sec (00:00:14) |  530 sec (00:08:50) |  175 sec (00:02:55) |    6 sec (00:00:06) |   22 sec (00:00:22) | 1139 sec (00:18:59) |   16 sec (00:00:16) | 1902 sec (00:31:42) |

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    <para><emphasis role="strong">Q</emphasis>: Will I have to login to
+    ConfigManager?</para>
+
+    <para><emphasis role="strong">A</emphasis>: No. (out of scope)</para>
```
out of scope?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    to login with?</para>
+
+    <para><emphasis role="strong">A</emphasis>: (TBD) Use your Risk
+    credentials.</para>
```
RISK credentials?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+	
+	<listitem>
+    <para><emphasis role="strong">Q</emphasis>: Can I now be logged
+    off?</para>
```
Can I log out of ECL Watch?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    off?</para>
+
+    <para><emphasis role="strong">A</emphasis>: Yes, you will be able to log
+    off, and if you do not your account will get locked after a configurable
```
Tell them how (menu option)

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+
+    <para><emphasis role="strong">A</emphasis>: No. After a configurable
+    period of inactivity your session will be locked. You will then need to
+    unlock to resume your session.</para>
```
This one could be combined with previous question

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+	<listitem>
+    <para><emphasis role="strong">Q</emphasis>: Will I still have to log into
+    ECLWatch in incognito mode, or delete the browser cookie cache, in order
+    to log in as a different user?</para>
```
no need to discuss previous hacks/workarounds. The previous question already said you can log in using different credentials in different tabs.  The two answers can be combined.

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+
+    <para><emphasis role="strong">A</emphasis>: No. You will not get logged
+    out, your session will get locked. Anything typed into any fields (such as
+    a search box) that has not been submit or entered could potentially be
```
has not been submitted

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    a search box) that has not been submit or entered could potentially be
+    lost. However, since the session is only locked, it is unlikely that any
+    data will be lost. The ECL IDE automatically saves your work so nothing
+    should get lost. </para>
```
Does the IDE lock?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+
+	      <listitem>
+    <para><emphasis role="strong">Q</emphasis>: Will HPCC command line
+    utilities such as testsocket or ecl be affected?</para>
```
testsocket is not a documented tool

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    utilities such as testsocket or ecl be affected?</para>
+
+    <para><emphasis role="strong">A</emphasis>: No. To the users they will
+    function exactly as before session management.</para>
```
rewrite : To the users they will function exactly as before session management.

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    reset the session expiration timer?</para>
+
+    <para><emphasis role="strong">A</emphasis>: No. Only user initiated
+    actions extend the timeout.</para>
```
duplicate Q and A

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+	
+	      <listitem>
+    <para><emphasis role="strong">Q</emphasis>: If I am logged in to the same
+    account using different browsers (ie Firefox and Chrome), do they share
```
i.e., means "that is" and  e.g., mean "for example."  Both need punctuation when you use them.
This sentence should say:  (e.g., Firefox and Chrome)

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+
+	
+	      <listitem>
+    <para><emphasis role="strong">Q</emphasis>: Can I automatically return to
```
comments offline

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    expired password from the login screen?</para>
+
+    <para><emphasis role="strong">A</emphasis>: Yes, this behaves just like it
+    currently does. You will be prompted to enter the old and new
```
use present tense

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+    to log in?</para>
+
+    <para><emphasis role="strong">A</emphasis>: Enter the same SDS credentials
+    you already use.</para>
```
SDS?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+
+    <para><emphasis role="strong">A</emphasis>: These secure protocols are
+    already available to be configured by an HPCC Administrator. Although not
+    required for session management, it is hoped that they are
```
it is hoped?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+
+	      <listitem>
+    <para><emphasis role="strong">Q</emphasis>: Will my programmatic SOAP
+    calls utilizing ESP have any impact from the sessions?</para>
```
impact from the sessions?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,269 @@
+     </listitem>
+
+	      <listitem>
+    <para><emphasis role="strong">Q</emphasis>: When will I see the Session
```
is this question valid since the docs are version 7?

JamesDeFabia reviewed (changes requested):
Comments inline

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: ff2b0f267b34525dfe4c525364829bab16b39138
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    7 sec (00:00:07) |  547 sec (00:09:07) |  167 sec (00:02:47) |    6 sec (00:00:06) |   21 sec (00:00:21) | 1110 sec (00:18:30) |   14 sec (00:00:14) | 1872 sec (00:31:12) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 34cc88c9f0e63bcf10dde9015bba6a489d45624e
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   12 sec (00:00:12) |  521 sec (00:08:41) |  183 sec (00:03:03) |    6 sec (00:00:06) |   22 sec (00:00:22) | 1116 sec (00:18:36) |   15 sec (00:00:15) | 1875 sec (00:31:15) |

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -86,18 +85,17 @@
-        account will get locked after a configurable period of
-        inactivity.</para>
+        logout available. You are able to log off, and if you do not your
+        account gets locked after a configurable period of inactivity.</para>
```
the account gets locked?

JamesDeFabia reviewed (changes requested):
comment inline. In general, this is much better

JamesDeFabia commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -64,49 +73,99 @@
+      <sect2>
+        <title>User Login</title>
+
+        <para>The first thing you encounter when starting a session in ECL
```
When you open ECL Watch, a log in window displays.

JamesDeFabia commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -64,49 +73,99 @@
+            </mediaobject>
+          </figure></para>
+
+        <para>You will login with the credentials supplied by your Systems
```
Present tense

JamesDeFabia commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -64,49 +73,99 @@
+
+        <para>You will login with the credentials supplied by your Systems
+        Administrator. After a configurable period of inactivity your ECL
+        Watch session will lock and you would need to enter your credentials
```
Active voice, present tense

JamesDeFabia reviewed (changes requested):
comments inline

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,255 @@
+        <para><emphasis role="strong">Q</emphasis>: What credentials should I
+        use to login with?</para>
+
+        <para><emphasis role="strong">A</emphasis>: (TBD) Use your assigned
```
Why is this TBD ?

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,255 @@
+        expires?</para>
+
+        <para><emphasis role="strong">A</emphasis>: This depends on your
+        system policies. If your password does expire, you will be presented a
```
"This depends on your system policies and the configured security manager"

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,255 @@
+
+        <para><emphasis role="strong">A</emphasis>: No. They function exactly
+        the same as before session management.</para>
+      </listitem>
```
This is only true if ESP is not configured for "AuthTypeSessionOnly."  Confirm with Kevin

RussWhitehead replied:
Still wrong

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,255 @@
+        ECLWatch reset the session expiration timer?</para>
+
+        <para><emphasis role="strong">A</emphasis>: No. Only user initiated
+        actions extend the timeout.</para>
```
Might want to mention somewhere that scrolling a window does NOT reset the timeout. Only mouse clicks and key presses

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,255 @@
+      </listitem>
+
+      <listitem>
+        <para><emphasis role="strong">Q</emphasis>: Which credentials should I
```
Didn't you ask this question above?

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,255 @@
+        calls utilizing ESP have any impact?</para>
+
+        <para><emphasis role="strong">A</emphasis>: No. Programmatic SOAP
+        calls continue to operate as they do now.</para>
```
Not true if "AuthTypeSessionOnly". Confirm with Kevin

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,255 @@
+        Session Management changes?</para>
+
+        <para><emphasis role="strong">A</emphasis>: Session Management will be
+        included as part of HPCC Version 7.0.</para>
```
.. and must be configured by your HPCC Admin

RussWhitehead replied:
Still wrong

RussWhitehead commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -29,8 +37,9 @@
-      of LexisNexis Risk Data Management Inc.</para>
+      <para><!-- DNT-Start -->HPCC<!-- DNT-End -->
+      Systems<superscript>®</superscript> is a registered trademark of
+      LexisNexis Risk Data Management Inc.</para>
```
I am not familiar with "LexisNexis Risk Data Management Inc".   Are we sure about this?

JamesDeFabia replied:
LexisNexis Risk Data Management Inc is correct. I just reconfirmed from the internal trademark list to be sure.

RussWhitehead commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -64,49 +73,98 @@
+        Administrator. After a configurable period of inactivity your ECL
+        Watch session locks and you need to enter your credentials to unlock
+        and resume your session. The session remains active for as long as
+        there is regular interaction.</para>
```
as long as there is regular keyboard and mouse interaction

RussWhitehead commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -64,49 +73,98 @@
+        there is regular interaction.</para>
+
+        <para>Additionally, there is a Logout menu item available you can
+        close your session when you complete your work.</para>
```
Additionally, there is a Logout menu item available you can "CLICK TO" close your session when you complete your work

RussWhitehead reviewed (changes requested):
Some comments

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: ba2985991794709a0f4c231c64fcbbf023975de3
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   17 sec (00:00:17) |  539 sec (00:08:59) |  169 sec (00:02:49) |    6 sec (00:00:06) |   21 sec (00:00:21) | 1145 sec (00:19:05) |   16 sec (00:00:16) | 1913 sec (00:31:53) |

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -173,8 +172,9 @@
-        actions extend the timeout.</para>
+        <para><emphasis role="strong">A</emphasis>: No. Only active
+        interactions like mouse clicks and pressing keys extend the timeout.
+        Note scrolling does not extend the expiration timer. </para>
```
What is note scrolling?

JamesDeFabia commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -138,10 +138,10 @@
         Watch session locks and you need to enter your credentials to unlock
         and resume your session. The session remains active for as long as
-        there is regular interaction.</para>
+        there is regular keyboard and mouse interaction.</para>
```
keyboard or mouse

JamesDeFabia commented on docs/EN_US/ECLWatch/TheECLWatchMan.xml:
```diff
@@ -138,10 +138,10 @@
 
         <para>Additionally, there is a Logout menu item available you can
-        close your session when you complete your work.</para>
+        click on and close your session when you complete your work.</para>
```
Additionally, there is a Logout menu item available you can select to close your session when you complete your work

JamesDeFabia reviewed (changes requested):
comments inline

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 74ea39373796bc106ca00596c256202a3c7771d4
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   20 sec (00:00:20) |  503 sec (00:08:23) |  171 sec (00:02:51) |    6 sec (00:00:06) |   22 sec (00:00:22) | 1124 sec (00:18:44) |   15 sec (00:00:15) | 1861 sec (00:31:01) |

g-pan commented:
@RussWhitehead  please review revisions

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 14747f3797aad6f2c90dcf33fc1105b29cc8c0a3
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    6 sec (00:00:06) |  556 sec (00:09:16) |  181 sec (00:03:01) |    6 sec (00:00:06) |   22 sec (00:00:22) | 1112 sec (00:18:32) |   15 sec (00:00:15) | 1898 sec (00:31:38) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 630c12552c2b58eb2a050c5dcde9a780fed0ea7f
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   31 sec (00:00:31) |  544 sec (00:09:04) |  175 sec (00:02:55) |    6 sec (00:00:06) |   21 sec (00:00:21) | 1113 sec (00:18:33) |   16 sec (00:00:16) | 1906 sec (00:31:46) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 8b08d5fb6da0824f99b52d545f73d8456131762c
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |85 |85 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   15 sec (00:00:15) |  512 sec (00:08:32) |  185 sec (00:03:05) |    6 sec (00:00:06) |   21 sec (00:00:21) | 1113 sec (00:18:33) |   15 sec (00:00:15) | 1867 sec (00:31:07) |

g-pan commented:
@RussWhitehead please review latest changes

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,250 @@
+
+      <listitem>
+        <para><emphasis role="strong">Q</emphasis>: Will I have to login to
+        ConfigManager?</para>
```
Configuration Manager

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,250 @@
+
+        <para><emphasis role="strong">A</emphasis>: Yes, there is a link to
+        logout available. You are able to log off, and if you do not your
+        account locks after a configurable period of inactivity.</para>
```
Your account doesn't lock

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,250 @@
+      <listitem>
+        <para><emphasis role="strong">Q</emphasis>: Will I be able to log in
+        as a different user?</para>
+
```
Duplicate question

g-pan replied:
Not really. This is very similar, but not the same.

JamesDeFabia replied:
Now that you have changed the order, it works

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,250 @@
+
+      <listitem>
+        <para><emphasis role="strong">Q</emphasis>: Is there any option to not
+        time out from inactivity?</para>
```
these two questions are almost the same and would be better combined

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,250 @@
+        <para><emphasis role="strong">A</emphasis>: Possibly. If you have
+        configured AuthPerSessionOnly then some command line utilities may not
+        function the same as before session management.</para>
+      </listitem>
```
Different how?

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,250 @@
+      <listitem>
+        <para><emphasis role="strong">Q</emphasis>: Does auto refresh in
+        ECLWatch reset the session expiration timer?</para>
+
```
Duplicate question

JamesDeFabia reviewed (changes requested):
comments inline

JamesDeFabia commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,250 @@
+        <para><emphasis role="strong">Q</emphasis>: Will I be able to change
+        an expired password from the login screen?</para>
+
+        <para><emphasis role="strong">A</emphasis>: Yes, this behaves just as
```
If your password is expired, you are redirected to a page that allows you to reset it

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: b96ea9cd050102af76669e54eb780ec7696b10ca
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |85 |85 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   30 sec (00:00:30) |  533 sec (00:08:53) |  175 sec (00:02:55) |    6 sec (00:00:06) |   22 sec (00:00:22) | 1121 sec (00:18:41) |   16 sec (00:00:16) | 1903 sec (00:31:43) |

g-pan commented:
@RussWhitehead please review   @JamesDeFabia please review

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,243 @@
+        <para><emphasis role="strong">Q</emphasis>: How long will an inactive
+        session last?</para>
+
+        <para><emphasis role="strong">A</emphasis>: You can configure this
```
An HPCC Admin can configure this....

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,243 @@
+        <para><emphasis role="strong">A</emphasis>: Possibly. If you have
+        configured AuthPerSessionOnly then command line utilities will not
+        work. If AuthPerSessionOnly is not enabled then command line utilities
+        will not be effected.</para>
```
Please confirm this with Kevin

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,243 @@
+      <listitem>
+        <para><emphasis role="strong">Q</emphasis>: If I am logged in to the
+        same account using multiple tabs in a browser, or multiple instances
+        of the same browser, can I get logged out of one but not the
```
How do you "get logged out"?

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,243 @@
+        out?</para>
+
+        <para><emphasis role="strong">A</emphasis>: Yes. The intent is to lock
+        your session and not completely log you out. Therefore logging back in
```
"logging back in" is inaccurate since you are not logged out.

RussWhitehead commented on docs/EN_US/ECLWatch/ECLWa_mods/SessFaq.xml:
```diff
@@ -0,0 +1,243 @@
+        <para><emphasis role="strong">A</emphasis>: Maybe. If you have
+        configured AuthPerSessionOnly then SOAP calls will not work. If your
+        system is not configured that way, then programmatic SOAP calls
+        continue to operate as they do now.</para>
```
Confirm with Kevin

g-pan committed:
HPCC-16262 DOCS:Session Management

Fix HPCC-16262 DOCS:Session Management
Document new HPCC Session Management...
also address HPCC-18904

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

g-pan commented:
@RussWhitehead please re-review. (all CLI statements confirmed with Kevin)

RussWhitehead commented:
@g-pan Looks good

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: c8c23d6b50c72798cb60206dc94a7d24dad5dac5
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |85 |85 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   13 sec (00:00:13) |  544 sec (00:09:04) |  160 sec (00:02:40) |    6 sec (00:00:06) |   21 sec (00:00:21) | 1129 sec (00:18:49) |   16 sec (00:00:16) | 1889 sec (00:31:29) |

JamesDeFabia reviewed (approved):
Looks good now
